dvc/tree/git.py

```diff
@@ -216,7 +216,11 @@ def to_ctime(git_time):
             return sec + nano_sec / 1000000000
 
         obj = self._git_object_by_path(path)
-        if obj is None:
+        if (
+            obj is None
+            or self.dvcignore.is_ignored_dir(path)
+            or self.dvcignore.is_ignored_file(path)
```

Comment:
Potential for optimization. Probably faster to check dvcignore than to `_git_object_by_path`